### ISSUE TITLE
Error Compiling modules from the shell in OTP25

### ISSUE THREAD

**Issue by lgmfred** [nitrogen/nitrogen#139]
I'm running nitrogen over Yaws on OTP 25 [erts-13.2.2]. I get the following error when I try to compile a module using either `c(module_name)` or `sync` ([the on-the-fly recompiling and reloading utility](https://github.com/rustyio/sync)). Running `make` and compiling from a normal (non-Nitrogen) shell works fine though.

```shell
(myapp@127.0.0.1)1> c(usage_statistics).
Recompiling /home/user/myapp/site/src/usage_statistics.erl
/home/user/myapp/site/src/usage_statistics.erl: internal error in pass parse_module:
exception error: undefined function proplists:to_map/1
  in function  compile:metadata_add_features/2 (compile.erl, line 1072)
  in call from compile:do_parse_module/2 (compile.erl, line 1038)
  in call from compile:parse_module/2 (compile.erl, line 994)
  in call from compile:fold_comp/4 (compile.erl, line 405)
  in call from compile:internal_comp/5 (compile.erl, line 389)
  in call from compile:'-internal_fun/2-anonymous-0-'/2 (compile.erl, line 227)
  in call from compile:'-do_compile/2-anonymous-0-'/1 (compile.erl, line 217)
error

(myapp@127.0.0.1)2> sync:go().
Starting Sync (Automatic Code Compiler / Reloader)
.
.
.
 =ERROR REPORT==== 24-Aug-2023::16:13:59.302768 ===
/home/user/myapp/site/src/usage_statistics.erl none: ☠ Error: internal error in pass parse_module:
exception error: undefined function proplists:to_map/1
  in function  compile:metadata_add_features/2 (compile.erl, line 1072)
  in call from compile:do_parse_module/2 (compile.erl, line 1038)
  in call from compile:parse_module/2 (compile.erl, line 994)
  in call from compile:fold_comp/4 (compile.erl, line 405)
  in call from compile:internal_comp/5 (compile.erl, line 389)
  in call from compile:'-internal_fun/2-anonymous-0-'/2 (compile.erl, line 227)
  in call from compile:'-do_compile/2-anonymous-0-'/1 (compile.erl, line 217)

=ERROR REPORT==== 24-Aug-2023::16:13:59.303569 ===
** gen_event handler yaws_log_file_h crashed.
** Was installed in error_logger
** Last event was: {error,<0.276.0>,
                          {<0.280.0>,
                           "/home/user/myapp/site/src/usage_statistics.erl none: ☠ Error: internal error in pass parse_module:\nexception error: undefined function proplists:to_map/1\n  in function  compile:metadata_add_features/2 (compile.erl, line 1072)\n  in call from compile:do_parse_module/2 (compile.erl, line 1038)\n  in call from compile:parse_module/2 (compile.erl, line 994)\n  in call from compile:fold_comp/4 (compile.erl, line 405)\n  in call from compile:internal_comp/5 (compile.erl, line 389)\n  in call from compile:'-internal_fun/2-anonymous-0-'/2 (compile.erl, line 227)\n  in call from compile:'-do_compile/2-anonymous-0-'/1 (compile.erl, line 217)\n",
                           []}}
** When handler state == {st,<0.145.0>,"./report.log",[],unlimited}
** Reason == {no_translation,
                 [{io,put_chars,
                      [<0.145.0>,
                       [[10,61,"ERROR REPORT",61,61,61,61,32,"24",45,"Aug",45,
                         "2023",58,58,"16",58,"13",58,"59",32,[],61,61,61,10],
                        [],
                        ["/home/user/myapp/site/src/usage_statistics.erl none: ☠ Error: internal error in pass parse_module:\nexception error: undefined function proplists:to_map/1\n  in function  compile:metadata_add_features/2 (compile.erl, line 1072)\n  in call from compile:do_parse_module/2 (compile.erl, line 1038)\n  in call from compile:parse_module/2 (compile.erl, line 994)\n  in call from compile:fold_comp/4 (compile.erl, line 405)\n  in call from compile:internal_comp/5 (compile.erl, line 389)\n  in call from compile:'-internal_fun/2-anonymous-0-'/2 (compile.erl, line 227)\n  in call from compile:'-do_compile/2-anonymous-0-'/1 (compile.erl, line 217)\n"]]],
                      [{file,"io.erl"},
                       {line,99},
                       {error_info,
                           #{cause => {device,{no_translation,unicode,latin1}},
                             module => erl_stdlib_errors}}]},
                  {error_logger_file_h,handle_event,2,
                      [{file,"error_logger_file_h.erl"},{line,71}]},
                  {gen_event,server_update,4,
                      [{file,"gen_event.erl"},{line,802}]},
                  {gen_event,server_notify,4,
                      [{file,"gen_event.erl"},{line,784}]},
                  {gen_event,handle_msg,6,[{file,"gen_event.erl"},{line,526}]},
                  {proc_lib,init_p_do_apply,3,
                      [{file,"proc_lib.erl"},{line,240}]}]}
```

**Comment by choptastic** [comment]
Hi @lgmfred,

Given that `proplists:to_map` was introduced in OTP24, but you're running OTP25, this looks like there's some kind of inconsistency in your system.

I would try to completely delete the `_build` directory, then re-run `make` and see what kicks out.

**Comment by choptastic** [comment]
Or wait, is this with `nitrogen 2.x` still, rather than Nitrogen 3 (a.k.a the `rebar3` branch?)

If that's the case, is this a full release or a slim release?

In either case, I would try this:

`make fix-slim-release` (if this is a full release, it will convert it to a slim release)

Then run `make`

If, after that is run, it continues to do that, I would then delete the `lib` directory and run `make` again and see what happens.

**Comment by lgmfred** [comment]
Yeah, there's some inconsistency. I'm running Nitrogen `2.4.0` on `Ubuntu 22.04 LTS`, and I've tried the above, but it didn't work either. I was using ASDF, but I even tried directly installing [esl-erlang](https://binaries2.erlang-solutions.com/ubuntu/pool/contrib/e/esl-erlang/esl-erlang_25.3.2-1~ubuntu~jammy_amd64.deb) package. However, the same error still pops up.

From the Nitrogen shell, the `proplists:to_map/2` is undefined. But, starting a normal Erlang shell from the same Nitrogen directory, the function `proplists:to_map/2` is defined and compiling works fine.

I just decided to revert my dev environment to use OTP24 instead.
